### PR TITLE
Build options

### LINKED ISSUE #10: Build variants for NAMD and LAMMPS
NAMD and LAMMPS have build variants (MPI) etc that we are currently not building. We should look into this.

### PR DESCRIPTION
**Pull request by ljwoods2**
Fixes #10 

Make GPU and MPI options available for simulation engines, but also create "vanilla" build for simple use in CI testing.

Separate simulation engines into different containers to make them lightweight

### CODE REVIEW

File: docker/lammps/env.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - zlib
+  - wget
+  - hdf5
+  - fftw
+  - tk
+  - hdf5
+  - curl
+  - openmp
+  - openmpi
+
```
Comment (hmacdope) [inline docker/lammps/env.yaml]: Whitespace changes

File: docker/namd/env.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - zlib
+  - wget
+  - hdf5
+  - fftw
+  - tk
+  - hdf5
+  - curl
+  - openmp
+  - openmpi
+
```
Comment (hmacdope) [inline docker/namd/env.yaml]: Whitespace changes

**Review by hmacdope — CHANGES_REQUESTED** [review]
I think this is a good idea to split them up for now, however we will likely still need a container with all three engines installed for tutorial / demos, where people will want to run with all three engines.

Similar for CI, will we want to pull in 3 different containers or only one? I'm not sure, open question. 

 Perhaps adding another directory with `common` that builds all three will solve that?

**Comment by ljwoods2** [comment]
Fair enough, my thinking is that for actual use after the workshop but before we actually merge this code into an i.e. gromacs release, people would want to have a container with just the simulation engine they want to use rather than a massive one. Though it makes sense to me to have one demo/test container where we dump everything. I can create a "common" one. Should this have all the different GPU/threading options as well? Or just GPU and nothing else?

For imdclient CI, we still need only-cpu build unless we get GPU runners somehow

**Comment by hmacdope** [comment]
@ljwoods2 

>  I can create a "common" one. Should this have all the different GPU/threading options as well? Or just GPU and nothing else?

For common I would do one common all GPU and one common all CPU with our most useful config for each (no MPI)

**Comment by ljwoods2** [comment]
@hmacdope Didn't end up doing a multi-stage build since this would've required installing another nearly-identical conda environment into the second stage without gcc, cmake, cudatools, etc (all the build specific dependencies) which felt redundant. This should be a "good enough" solution that makes the containers lighter by a few GB.

**Comment by ljwoods2** [comment]
Going to merge and test this, made a backup of main on "archive" branch in case we need to hard reset for some reason

**Pull request merged**